hubmap/query_app/filters.py

```diff
@@ -82,29 +89,30 @@ def process_single_condition(
     Finds the keyword args for a quantitative query based on the results of
     calling split_at_comparator() on a string representation of that condition"""
 
+    cell_dfs_dict = {"atac": atac_cell_df, "codex": codex_cell_df, "rna": rna_cell_df}
+
     value = float(split_condition[2].strip())
 
     var_id = split_condition[0].strip()
 
     if input_type == "protein":
         modality = "codex"
-        adata = codex_adata
     elif genomic_modality == "rna":
         modality = "rna"
-        adata = rna_adata
     elif genomic_modality == "atac":
         modality = "atac"
-        adata = atac_adata
 
-    if var_id not in adata.var.index:
-        raise ValueError(f"{var_id} not present in {modality} index")
+    cell_df = cell_dfs_dict[modality]
 
-    adata = adata[:, [var_id]]
-    bool_series = adata.X > value
-    adata = adata[bool_series, :]
-    cell_ids = list(adata.obs.index)
+    try:
+        operator = operators_dict[split_condition[1].strip()]
+        num_array = zarr_root[f"/{modality}/{var_id}"][:]
+        bool_array = operator(num_array, value)
+        cell_ids = cell_df[bool_array].cell_id.to_list()
+        return Q(cell_id__in=cell_ids)
 
-    return Q(cell_id__in=cell_ids)
+    except Exception as e:
```

Comment:
What specific types of exceptions should be caught here? Should catch the most specific exception possible.